backend/oasst_backend/api/v1/auth.py

```diff
@@ -0,0 +1,46 @@
+from typing import Union
+
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+from fastapi import APIRouter, Depends, Security
+from fastapi.security import APIKeyCookie
+from jose import jwe
+from oasst_backend.config import settings
+from pydantic import BaseModel, EmailStr
+
+router = APIRouter()
+
+oauth2_scheme = APIKeyCookie(name=settings.AUTH_COOKIE_NAME)
+
+
+class TokenData(BaseModel):
+    """
+    A minimal re-creation of the web's token type.  To be expanded later.
+    """
+
+    email: Union[EmailStr, None] = None
+
+
+async def get_current_user(token: str = Security(oauth2_scheme)):
+    """
+    Decrypts the user's JSON Web Token using HKDF encryption and returns the
+    TokenData.
+    """
+    # We first generate a key from the auth secret.
+    hkdf = HKDF(
+        algorithm=hashes.SHA256(),
+        length=settings.AUTH_LENGTH,
+        salt=settings.AUTH_SALT,
+        info=settings.AUTH_INFO,
+    )
+    key = hkdf.derive(settings.AUTH_SECRET)
```

Comment:
these are constant for the entire lifetime of the application, correct? can we move them out of the function?

Reply:
I tried that.  HKDR didn't properly generate the key.  I'd like to keep this as is since I copied it from how NextAuthJS does their JWE encryption/decryption.

Reply:
ok, if it didn't work then there is some magic happening that I don't understand.

Reply:
The cryptography library described this as dangerous like Dinosaurs with Laser guns.  So there's definitely some magic we should be careful about.